src/main/java/io/confluent/connect/hdfs/DataWriter.java

```diff
@@ -97,6 +99,17 @@ public DataWriter(HdfsSinkConnectorConfig connectorConfig, SinkTaskContext conte
         conf.addResource(new Path(hadoopConfDir + "/hdfs-site.xml"));
       }
 
+      final String formatClass = connectorConfig.getString(HdfsSinkConnectorConfig.FORMAT_CLASS_CONFIG);
+      final String formatClassCompression = connectorConfig.getString(HdfsSinkConnectorConfig.FORMAT_CLASS_COMPRESSION_CONFIG);
+      if (StringUtils.isNotEmpty(formatClassCompression)) {
+          if (!formatClass.equals("io.confluent.connect.hdfs.avro.AvroFormat")) {
```

Comment:
Could we just have the `Format` implementations look up this config themselves instead? Checking for the format class seems a bit hacky, and since it's pluggable, making this fail for everything other than `AvroFormat` if the config is provided would mean a user wouldn't be able to implement their own `Format` that supports this config.